service/src/main/java/tds/assessment/web/endpoints/ItemController.java

```diff
@@ -0,0 +1,50 @@
+package tds.assessment.web.endpoints;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Optional;
+
+import tds.assessment.ItemFileMetadata;
+import tds.assessment.services.ItemService;
+import tds.common.web.exceptions.NotFoundException;
+
+@RestController
+public class ItemController {
+    private final ItemService itemService;
+
+    @Autowired
+    public ItemController(final ItemService itemService) {
+        this.itemService = itemService;
+    }
+
+    @GetMapping(value = "assessments/item/metadata", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseBody
+    ResponseEntity<ItemFileMetadata> findItemFileMetadata(@RequestParam final String clientName,
+                                                          @RequestParam final long bankKey,
+                                                          @RequestParam(required = false) final Long stimulusKey,
+                                                          @RequestParam(required = false) final Long itemKey) {
+        if (stimulusKey == null && itemKey == null) {
```

Comment:
We only care if both are not provided since one or the other is required.  If there is a better way to construct the URL we can tweak it.